src/promptflow/promptflow/core/_utils.py

```diff
@@ -110,6 +121,269 @@ def generate_flow_meta(
     return flow_meta
 
 
+def render_jinja_template_content(template_content, *, trim_blocks=True, keep_trailing_newline=True, **kwargs):
+    template = Template(template_content, trim_blocks=trim_blocks, keep_trailing_newline=keep_trailing_newline)
+    return template.render(**kwargs)
+
+
+def get_connection(connection):
+    if not isinstance(connection, (str, dict)):
+        error_message = """
+        Illegal definition of connection. Need to provide connection name or connection info like below:
+        connection:
+            type: <connection_type>
+            api_key: <api_key>
+            api_base: <api_base>
+            ...
+        """
+        raise InvalidConnectionError(message=error_message)
+    if isinstance(connection, str):
+        # Get connection by name
+        try:
+            from promptflow._sdk._pf_client import PFClient
+        except ImportError as ex:
+            raise CoreError(f"Please try 'pip install promptflow-devkit' to install dependency, {ex.msg}")
+        client = PFClient()
+        connection_obj = client.connections.get(connection, with_secrets=True)
+        connection = connection_obj._to_execution_connection_dict()["value"]
+        connection_type = connection_obj.TYPE
+    else:
+        connection_type = connection.pop("type", None)
+    if connection_type == AzureOpenAIConnection.TYPE:
+        return AzureOpenAIConnection(**connection)
+    elif connection_type == OpenAIConnection.TYPE:
+        return OpenAIConnection(**connection)
+    error_message = (
+        f"Not Support connection type {connection_type} for embedding api. "
+        f"Connection type should be in [{AzureOpenAIConnection.TYPE}, {OpenAIConnection.TYPE}]."
+    )
+    raise InvalidConnectionTypeError(message=error_message)
+
+
+# region: Copied from promptflow-tools
+
+
+def normalize_connection_config(connection):
```

Comment:
shall we move new utils to core/_prompty_utils.py?

Reply:
move to _prompty_utils.py